### PR TITLE
feat(api): adopt the impl Into<Jid> convention across the public surface

### PR DESCRIPTION
**Pull request by jlucaso1**
## Problem

The four most-used methods of the library (`send_message`, `edit_message`, `pin/unpin_message`, `revoke_message`) took `Jid` by value while the entire feature surface took `&Jid`. The common caller keeps the jid and sends repeatedly, so every call site read `client.send_message(jid.clone(), msg)`; the library itself paid the inverse cost, with 10 internal `send_message(chat.clone(), ...)` sites because features hold references. Two conventions on one surface, and the noisy one sat on the hottest entry points. Pre-1.0 is the last cheap moment to fix the signature of the methods every consumer touches.

## Change

One convention: **public methods that end up owning their `Jid` take `impl Into<Jid>`**, backed by a new `From<&Jid> for Jid` impl.

- Callers with a reference call `client.send_message(&jid, msg)` and pay one clone inside (the user part is `CompactString`-inline for typical numeric ids, so no heap).
- Callers done with the jid pass it by value and move for free.
- The 10 internal forced clones are gone.

**Monomorphization discipline:** the large async bodies (`send_message_with_options`, `edit_message`, `edit_message_encrypted`, `revoke_message`) keep a monomorphic `_inner` behind a thin generic shim that only does `.into()`, so `Into` instantiations cannot duplicate the send state machines. Small delegating methods take the generic directly. The internal `send_message_impl` stays monomorphic on purpose.

**Scope (classified by an adversarially-verified sweep of the whole public surface):** ~53 methods converted across the messaging core and the owning feature surface (`groups` 26, `community` 6, `newsletter` 2, `presence` 1, `events`, `polls`, `comments`, `reactions`, `keep_message`). Deliberate exceptions:

- Read-only methods keep `&Jid`: converting them would force a clone on borrow-callers. The clearest case the verification caught is `groups().query_info`, whose cache hit only reads the jid on the per-send hot path; ~59 read-only methods stay as they are.
- `&[Jid]` slices and `Option<&Jid>` params are out of scope.
- HKDF identity params that are only read (`event_creator_jid`, `poll_creator_jid`) stay `&Jid`.

## Compatibility

Call sites passing owned `Jid` compile unchanged. Call sites that previously had to clone can drop the clone. Feature methods that took `&Jid` keep accepting `&jid` through the `From<&Jid>` impl, so this is source-compatible for the dominant call style; pre-1.0 breaking surface is limited to type-inference edge cases (e.g. passing `.parse()?` results now needs no change either, since `Jid: Into<Jid>`).

## Tests

- A never-executed compile guard (`jid_into_convention`) pins BOTH call styles (`&jid` and owned) for every core method, so removing `From<&Jid>` or regressing a signature fails the build.
- Full workspace suite green (2165 tests), `cargo clippy --all-targets -- -D warnings` clean, both wasm32 CI variants reproduced locally.

## Breaking

Signatures of the converted methods change from `Jid`/`&Jid` to `impl Into<Jid>` (source-compatible for owned and `&jid` callers; trait-object/fn-pointer uses of these inherent methods, if any exist downstream, would need a closure). Pre-1.0.

### CODE REVIEW

File: src/send.rs [hunk 1]
```diff
@@ -4682,3 +4709,34 @@ mod tests {
...
+        let _ = client
+            .pin_message(&jid, wa::MessageKey::default(), PinDuration::default())
+            .await;
+        let _ = client.unpin_message(&jid, wa::MessageKey::default()).await;
+        let _ = client
+            .send_reaction(&jid, wa::MessageKey::default(), "x")
+            .await;
+        // Owned style: moves, no clone.
+        let _ = client.send_message(jid, msg).await;
+    }
```
[the remark this answers is not in the document]
Comment (jlucaso1) [reply src/send.rs:4771]: Applied in the latest push: every core method now also consumes an owned Jid copy in the guard, so both styles are pinned across the whole surface.

File: src/send.rs [hunk 2]
```diff
@@ -4682,3 +4709,61 @@ mod tests {
...
+            .await;
+        let _ = client
+            .unpin_message(jid.clone(), wa::MessageKey::default())
+            .await;
+        let _ = client
+            .send_reaction(jid.clone(), wa::MessageKey::default(), "x")
+            .await;
+        let _ = client
+            .keep_message(jid, wa::MessageKey::default(), true)
+            .await;
```
[the remark this answers is not in the document]
Comment (jlucaso1) [reply src/send.rs:4770]: Done in the latest push, keep_message now appears on both styles like the rest.
